coder-sdk/request.go

```diff
@@ -4,51 +4,62 @@ import (
 	"bytes"
 	"context"
 	"encoding/json"
+	"io"
 	"net/http"
 
 	"golang.org/x/xerrors"
 )
 
-func (c Client) request(
-	ctx context.Context,
-	method string, path string,
-	request interface{},
-) (*http.Response, error) {
-	client, err := c.http()
+// request is a helper to set the cookie, marshal the payload and execute the request.
+func (c Client) request(ctx context.Context, method, path string, in interface{}) (*http.Response, error) {
+	// Create a default http client with the auth in the cookie.
+	client, err := c.newHTTPClient()
 	if err != nil {
-		return nil, err
+		return nil, xerrors.Errorf("new http client: %w", err)
 	}
-	if request == nil {
-		request = []byte{}
-	}
-	body, err := json.Marshal(request)
-	if err != nil {
-		return nil, xerrors.Errorf("marshal request: %w", err)
+
+	// If we have incoming data, encode it as json.
+	var payload io.Reader
+	if in != nil {
+		body, err := json.Marshal(in)
+		if err != nil {
+			return nil, xerrors.Errorf("marshal request: %w", err)
+		}
+		payload = bytes.NewReader(body)
 	}
-	req, err := http.NewRequestWithContext(ctx, method, c.BaseURL.String()+path, bytes.NewReader(body))
+
+	// Create the http request.
+	req, err := http.NewRequestWithContext(ctx, method, c.BaseURL.String()+path, payload)
 	if err != nil {
 		return nil, xerrors.Errorf("create request: %w", err)
 	}
+	// TODO: Consider setting content-type / encoding / accept headers.
+
+	// Execute the request.
```

Comment:
I don't think it's necessary to have comments to describe function calls.

also, omit the todo or opening a ticket?

Reply:
I am not sure everyone makes the relation between `req.Do` and effectively executing the request, especially when having `http.NewRequest` earlier. Having comments to explain the flow in plain English makes things easier to follow and maintain.

If you think it is just too much, I"ll remove it.

Reply:
I'm still for removing. 

Comments are just one more thing that can get crufty, and while I agree that the interface is intuitive, I would hate that we should sprinkle descriptions of other package's APIs throughout calling code.